lib/errgonomic.rb

```diff
@@ -184,21 +275,45 @@ class Arbitrary; end
 
       attr_accessor :value
 
+      # Err may be constructed without a value, if you want.
+      #
+      # @example
+      #   Err("foo").value # => "foo"
+      #   Err().value # => Arbitrary
       def initialize(value = Arbitrary)
         @value = value
       end
 
+      # Err is always err
+      #
+      # @example
+      #   Err("foo").err? # => true
       def err?
         true
       end
 
+      # Err is never ok
+      #
+      # @example
+      #   Err("foo").ok? # => false
       def ok?
         false
       end
     end
 
     # This is my first stab at a basic DSL for matching and responding to
     # different Result variants.
+    #
+    # @example
+    #   Err("foo").match do
+    #     ok { :ok }
+    #     err { :err }
+    #   end # => :err
+    #
+    #   Ok("foo").match do
+    #     ok { :ok }
+    #     err { :err }
+    #   end # => :ok
```

Comment:
Do you support pattern matching? https://docs.ruby-lang.org/en/3.0/syntax/pattern_matching_rdoc.html Could you?

Reply:
great idea! added

```ruby
measurement = Some(1)
case measurement
in Some, value
  "Measurement is #{measurement.value}"
in None
  "Measurement is not available"
else
  "not matched"
end # => "Measurement is 1"
```

(plus or minus some constant namespacing shenanigans…)